pypeit/scripts/coadd_1dspec.py

```diff
@@ -83,6 +84,34 @@ def read_coaddfile(ifile):
     return cfg_lines, spec1dfiles, objids
 
 
+def build_coadd_file_name(spec1dfiles, spectrograph):
+    """Build the output file name for coadding.
+    The filename convention is coadd1d_<target>_<instrument name>_<YYYYMMDD>.fits or
+    coadd1d_<target>_<instrument name>_<YYYYMMDD>-<YYYYMMDD>.fits if the coadd included more than
+    one day's worth of data. The default location of the file will be along side the first spec1d file.
+
+    Currently instrument_name is taken from spectrograph.camera
+
+    Returns: 
+        str:  The name of the coadd output file.
+    """
+    mjd_list = [float(fits.getheader(f)['MJD']) for f in spec1dfiles]
```

Comment:
I'm concerned about how this is handled if for some reason the `MJD` keyword is missing. I don't remember if `fits.getheader(f)['MJD']` will give an error if `MJD` is not present or it will give you a `None`. I guess in both cases it would be good to have a way to handle it.

Reply:
This is an excellent point, and I am glad you caught this @deborape!

If the MJD does not exist, astropy.io.fits will throw an exception. 

`KeyError:` "Keyword 'MJD' not `found."`

So, instead of a list comprehension, that needs a for loop with a try/except. If the except is caught, I think there are a couple of choices. 

If there is some sort of date in the files, you can use 
`astropy.time.Time`
to convert the ISOT date string in the header to a MJD (this is two lines of code that I seem to use a lot.) 

The other option would be to punt on a date at all and use the KOA filenames (which are date based.)